src/frontend/src/optimizer/plan_node/stream_sink.rs

```diff
@@ -259,7 +259,12 @@ impl StreamSink {
                 );
             }
         };
-
+        // For file sink, it must have sink_decouple turned on.
+        if !sink_decouple && sink.is_fs_sink() {
+            return Err(
+                SinkError::Config(anyhow!("File sink cannot set sink_decouple to false.")).into(),
```

Comment:
"File sink can only be created with sink_decouple enabled. Please run `set sink_decouple to true` first."